`Multi::zipFilled()`, `Stream::zipFilledWith()` methods added

Smoren opened the pull request:
Hi @markrogoyski,

New methods:
* `Multi::zipFilled()`;
* `Stream::zipFilledWith()`.

Smoren committed:
`Multi::zipFilled()`, `Stream::zipFilledWith()` methods added and tested.

markrogoyski commented on src/Multi.php:
```diff
@@ -58,6 +59,30 @@ public static function zipLongest(iterable ...$iterables): \Generator
+     */
+    public static function zipFilled($filler, iterable ...$iterables): \Generator
+    {
+        $iterator = new JustifyMultipleIterator(...$iterables);
+        $iterator->setFiller($filler);
```
Suggestion: Perhaps consider a new "Filled" iterator or something that would take the filler as a constructor parameter along with the iterables. Using a setter mutates the iterator. It is generally a good idea to consider objects as immutable whenever possible. While an iterator technically does have mutable state with its internal pointer, we can avoid additional unnecessary mutations by requiring all parameters at the time of construction.

Smoren replied:
Done.

markrogoyski commented:
Hi @Smoren,

Thank you for the PR implementing zipFilled. Just one small suggestion about how to construct the iterator.

Thanks,
Mark

Smoren committed:
`JustifyMultipleIterator`: constructor interface extended with `$filler` param, method `setFiller()` removed.

Smoren commented:
Thank you @markrogoyski,

I've added `$filler` param to the constructor interface of `JustifyMultipleIterator` class and removed `setFiller()` method.
